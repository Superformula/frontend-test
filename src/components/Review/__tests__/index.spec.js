import React from 'react';
import { shallow } from 'enzyme';
import { Review } from '../';

describe('Review', () => {
  it('should render without failing', () => {
    const wrapper = shallow(<Review />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a complete review', () => {
    const review = {
      rating: 5,
      date: '01/01/2021',
      text: 'lorem ipsum',
      user: { name: 'tester', avatar: 'http://path.to/image.png' },
    };
    const wrapper = shallow(<Review {...review} />);
    expect(wrapper).toMatchSnapshot();
  });
});
