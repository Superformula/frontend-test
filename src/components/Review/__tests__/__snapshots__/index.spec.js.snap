// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review should render a complete review 1`] = `
<Wrapper>
  <UserWrapper>
    <ImageWrapper>
      <FitImg
        alt="tester"
        src="http://path.to/image.png"
      />
    </ImageWrapper>
    <div>
      <Name>
        tester
      </Name>
      <PostedAt>
        01/01/2021
      </PostedAt>
    </div>
  </UserWrapper>
  <div>
    <Rating
      $lg={false}
      $xs={false}
      max={5}
      rating={5}
    />
    <Paragraph
      extraWeight={true}
    >
      lorem ipsum
    </Paragraph>
  </div>
</Wrapper>
`;

exports[`Review should render without failing 1`] = `
<Wrapper>
  <UserWrapper>
    <ImageWrapper>
      <FitImg />
    </ImageWrapper>
    <div>
      <Name />
      <PostedAt />
    </div>
  </UserWrapper>
  <div>
    <Rating
      $lg={false}
      $xs={false}
      max={5}
      rating={0}
    />
    <Paragraph
      extraWeight={true}
    />
  </div>
</Wrapper>
`;
